test/e2e/framework/helpers/k8s/k8s_utils.go

```diff
@@ -1779,3 +1755,83 @@ func (k *KubeCtl) DeleteStorageClass(scName string) error {
 	}
 	return nil
 }
+
+func (k *KubeCtl) GetSecrets(namespace string) (*v1.SecretList, error) {
+	return k.clientSet.CoreV1().Secrets(namespace).List(context.TODO(), metav1.ListOptions{})
+}
+
+// GetSecretValue retrieves the value for a specific key from a Kubernetes secret.
+func (k *KubeCtl) GetSecretValue(namespace, secretName, key string) (string, error) {
+	secret, err := k.getSecretWithRetry(namespace, secretName)
+	if err != nil {
+		return "", err
+	}
+	// Check if the key exists in the secret
+	value, ok := secret.Data[key]
+	if !ok {
+		return "", fmt.Errorf("key %s not found in secret %s", key, secretName)
+	}
+	return string(value), nil
+}
+
+// getSecretWithRetry retries getting the secret until it's populated with data, up to a limit.
+func (k *KubeCtl) getSecretWithRetry(namespace, secretName string) (*v1.Secret, error) {
+	var secret *v1.Secret
+	var err error
+
+	for i := 0; i < 5; i++ {
+		secret, err = k.clientSet.CoreV1().Secrets(namespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		// If secret contains data, return it
+		if len(secret.Data) > 0 {
+			return secret, nil
+		}
+
+		// Wait before retrying
+		time.Sleep(2 * time.Second)
+	}
+
+	return nil, fmt.Errorf("secret %s has no data after retries", secretName)
+}
+
+func WriteConfigToFile(config *rest.Config, kubeconfigPath string) error {
```

Comment:
I'm OK with this for the time being, let's not waste time. We can figure out later if needed.